# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/fetch/BodyExtractor.cpp
# Commit: 1954ce9669b8
# Full Hash: 1954ce9669b86b61c418c8286fbe0754ad2419cd
# Author: Adam Vandolder <avandolder@mozilla.com>
# Date: 2024-05-15 20:24:18
# Regressor Bug: 1890428
# File Overlap Count: 1
# Description:
#   Bug 1890428 - Add serialization support for Declarative Shadow DOM. r=webidl,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D206955
# ==============================================================================

diff -r 3e45e3d15eed -r 1954ce9669b8 dom/fetch/BodyExtractor.cpp
--- a/dom/fetch/BodyExtractor.cpp	Wed May 15 03:01:18 2024 +0000
+++ b/dom/fetch/BodyExtractor.cpp	Wed May 15 03:01:19 2024 +0000
@@ -7,6 +7,7 @@
 #include "BodyExtractor.h"
 #include "mozilla/dom/File.h"
 #include "mozilla/dom/FormData.h"
+#include "mozilla/dom/ShadowRoot.h"
 #include "mozilla/dom/TypedArray.h"
 #include "mozilla/dom/URLSearchParams.h"
 #include "mozilla/dom/XMLHttpRequest.h"
@@ -87,7 +88,8 @@
     aContentTypeWithCharset.AssignLiteral("text/html;charset=UTF-8");
 
     nsString serialized;
-    if (!nsContentUtils::SerializeNodeToMarkup(mBody, true, serialized)) {
+    if (!nsContentUtils::SerializeNodeToMarkup(mBody, true, serialized, false,
+                                               {})) {
       return NS_ERROR_OUT_OF_MEMORY;
     }
 